widgetssdk/src/test/java/com/glia/widgets/callvisualizer/ActivityWatcherControllerTest.kt

```diff
@@ -203,22 +216,102 @@ internal class ActivityWatcherControllerTest {
+    }
+
+    @Test
+    fun `onMediaUpgradeAccept does destroy support activity when requestCameraPermission is true`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onMediaUpgradeAccept(null)
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline does cleanup when decline`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onMediaUpgradeDecline(null)
+        verify(watcher).dismissAlertDialog(true)
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline does not destroy support activity when error occurs`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onMediaUpgradeDecline(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+    }
+
+    @Test
+    fun `onInitialCameraPermissionResult accepts offer when permission is granted`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onInitialCameraPermissionResult(isGranted = true, isComponentActivity = true)
+    }
+
+    @Test
+    fun `onInitialCameraPermissionResult requests permission when permission is not granted and is ComponentActivity`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onInitialCameraPermissionResult(isGranted = false, isComponentActivity = true)
+        verify(watcher).requestCameraPermission()
+    }
+
+    @Test
+    fun `onInitialCameraPermissionResult opens ComponentActivity when permission is not granted and is not ComponentActivity`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onInitialCameraPermissionResult(isGranted = false, isComponentActivity = false)
+        verify(watcher).openComponentActivity()
+    }
+
+    @Test
+    fun `onRequestedCameraPermissionResult accepts when isGranted`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onRequestedCameraPermissionResult(true)
+        verify(watcher).destroySupportActivityIfExists()
+    }
+
+    @Test
+    fun `onRequestedCameraPermissionResult declines when !isGranted`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer)
+        resetMocks()
+        controller.onRequestedCameraPermissionResult(false)
+        verify(watcher).destroySupportActivityIfExists()
+    }
+
     private fun prepareMediaUpgradeApplicationState() {
         val offer = mock(MediaUpgradeOffer::class.java)
         val state = DialogState.MediaUpgrade(offer, "name", MODE_MEDIA_UPGRADE)
         controller.onDialogControllerCallback(state)
         verify(watcher).showUpgradeDialog(state)
     }
 
-    private fun `onActivityResumed callbacks are set when call or chat are not active`() {
+    private fun `onActivityResumed callbacks are set when call or chat are not active`(activity: Activity) {
```

Comment:
this function looks confusing to me, especially when it is called inside another function, I think we should rename it and also make it look like an ordinary function smth like 
`fun mockOnActivityResultAndEnsureCallbacksSet()`